src/importer/sources/retroarch_source.py

```diff
@@ -0,0 +1,135 @@
+# retroarch_source.py
+#
+# Copyright 2023 Rilic
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import hashlib
+import json
+import logging
+from json import JSONDecodeError
+from time import time
+
+from src import shared
+from src.errors.friendly_error import FriendlyError
+from src.game import Game
+from src.importer.sources.location import Location
+from src.importer.sources.source import Source, SourceIterationResult, SourceIterator
+
+
+class RetroarchSourceIterator(SourceIterator):
+    source: "RetroarchSource"
+
+    def generator_builder(self) -> SourceIterationResult:
+        added_time = int(time())
+
+        # Get all playlist files, ending in .lpl
+        playlist_files = self.source.config_location["playlists"].glob("*.lpl")
+
+        bad_playlists = set()
+
+        for playlist_file in playlist_files:
+            try:
+                with open(
```

Comment:
This can be simplified even more
```py
try:
    with playlist_file.open(encoding="utf-8") as open_file:
        playlist_json = json.load(open_file)
except ...
```